trie/branchNode.go

```diff
@@ -754,25 +697,31 @@ func (bn *branchNode) deleteChild(
 	hasBeenModified.SetValue(true)
 }
 
-func (bn *branchNode) setNewChild(childPos byte, newNode node) error {
+func (bn *branchNode) setNewChild(childPos byte, newNode node, modifiedHashes common.AtomicBytesSlice) error {
 	bn.mutex.Lock()
 	defer bn.mutex.Unlock()
 
+	if !bn.dirty && len(bn.hash) != 0 {
+		modifiedHashes.Append([][]byte{bn.hash})
+	}
+
 	bn.hash = nil
+	bn.dirty = true
```

Comment:
if mutex protection is added in baseNode, cand we use setters here and avoid mutex in this file entirely?